packages/core/src/components/size-attribute.ts

```diff
@@ -0,0 +1,82 @@
+import { DOM, customAttribute, bindingMode, inject, bindable } from 'aurelia-framework';
+import { PLATFORM } from 'aurelia-pal';
+
+export interface Size {
+  width: number;
+  height: number;
+}
+
+@customAttribute('size')
+@inject(Element)
+export class SizeCustomAttribute {
+
+  private observer: any;
+  @bindable({defaultBindingMode: bindingMode.fromView, primaryProperty: true})
+  public value: Size = {width: 0, height: 0};
+
+  @bindable({defaultBindingMode: bindingMode.fromView}) public width: number = 0;
+  @bindable({defaultBindingMode: bindingMode.fromView}) public height: number = 0;
+
+  constructor(private element: HTMLElement) {
+  }
+
+  public bind() {
+    this.observer = this.getObserver();
+    this.observer.observe(this.element);
+  }
+
+  public unbind() {
+    this.observer.disconnect();
+    this.observer = void 0;
+  }
+
+  public getObserver() {
+    if (typeof PLATFORM.global.ResizeObserver === 'function') {
+      return new PLATFORM.global.ResizeObserver((records: Array<{contentRect: DOMRectReadOnly}>) => {
+        const rect = records[0].contentRect;
+        this.sizeChanged({ width: rect.width, height: rect.height });
```

Comment:
`sizeChanged` will be call whenever underlying `size` property has been changed, I think we can avoid doing this manually.